### DISCUSSION TITLE
Abstracting away math backend

### DISCUSSION THREAD

**Discussion by timofeymukha** [ExtremeFLOW/neko#810] [General]
Is it possible to abstract away the backed for `math` in the same way we do other stuff? I.e. instead of forcing stuff like this 
```fortran
      if (NEKO_BCKND_DEVICE .eq. 1) then
         call device_add2(p%x_d, dp%x_d,n)
      else
         call add2(p%x, dp%x,n)
      end if
```
have an `add2`, which then has an if like that inside and calls the appropriate backed function. I can't see why not.
Is this #710 ?

**Comment by njansson** [comment]
Yes this is what #710 is about, and the main showstopper is tracked in #706. However, since very few routines (if any) in the solver uses the math interface, maybe the performance impact is less severe.

**Reply by timofeymukha** [reply]
I see, that's a bit tricky then. But would be very nice to have.
